src/modules/campath/mod.rs

```diff
@@ -0,0 +1,350 @@
+                    }
+                    None => done = true,
+                }
+            }
+        }
+    }
+
+    if done {
+        reset(marker);
+    }
+}
+
+static BXT_CAMPATH_EXPORT_START: Command = Command::new(
+    b"bxt_campath_export_start\0",
+    handler!(
+        "bxt_campath_export_start [filename.cam]
+
+Starts capturing player's position and viewangle from either gameplay or demo into .cam format.
+
+When capturing motion from demo, in order to speed up, try `timedemo \"demoname\"`.",
+        export_start as fn(_),
+        export_start_with_filename as fn(_, _)
+    ),
+);
+
+fn export_start(marker: MainThreadMarker) {
+    export_start_with_filename(marker, "output.cam".to_string());
+}
+
+fn export_start_with_filename(marker: MainThreadMarker, filename: String) {
+    if !Campath.is_enabled(marker) {
+        return;
+    }
+
+    if !filename.ends_with(".cam") {
+        con_print(marker, "Error: File name must end with \".cam\".\n");
+        return;
+    }
+
+    // TODO: make export possible when motion load exists while there's still more.
+    // Look at the comment in on_cl_disconnect()
+    if !matches!(*STATE.borrow_mut(marker), State::Idle) {
+        // Do not capture while loaded or exporting.
+        con_print(marker, "Error: Currently loaded or exporting");
+        return;
+    }
+
+    con_print(
+        marker,
+        &format!("Recording player motion into {}.\n", &filename),
+    );
+
+    reset(marker);
+    *STATE.borrow_mut(marker) = State::Exporting(Exporter::new(filename));
+}
+
+pub fn capture_motion(marker: MainThreadMarker) {
+    // Because R_RenderView is called multiple times before a frame is processed,
+    // it is better to hook this with V_RenderView or something else.
+    if !Campath.is_enabled(marker) {
+        return;
+    }
+
+    if matches!(*STATE.borrow(marker), State::Idle) {
+        return;
+    }
+
+    if let State::Exporting(ref mut exporter) = *STATE.borrow_mut(marker) {
+        let r_refdef_vieworg = unsafe { &mut *engine::r_refdef_vieworg.get(marker) };
+        let r_refdef_viewangles = unsafe { &mut *engine::r_refdef_viewangles.get(marker) };
+        let fov = unsafe { *engine::scr_fov_value.get(marker) };
+
+        exporter.append_entry(
+            TIME.get(marker),
+            [
+                r_refdef_vieworg[0] as f64,
+                r_refdef_vieworg[1] as f64,
+                r_refdef_vieworg[2] as f64,
+            ],
+            [
+                r_refdef_viewangles[2] as f64, // flip the order
+                r_refdef_viewangles[0] as f64,
+                r_refdef_viewangles[1] as f64,
+            ],
+            fov as f64,
+        );
+    }
+}
+
+static BXT_CAMPATH_EXPORT_STOP: Command = Command::new(
+    b"bxt_campath_export_stop\0",
+    handler!(
+        "bxt_campath_export_stop
+        
+Stops capturing player motion. Automatically stops when demo stops.",
```

Comment:
```suggestion
Stops capturing player motion.",
```
If anything, this should go into export_start docs.